### PR TITLE
Add validation for homepage protocols

### PR DESCRIPTION
**Pull request by brenetic**
- validation rake tasks now check for missing protocols in the homepage URL

### CODE REVIEW

File: lib/transition-config/site.rb [hunk 1]
```diff
@@ -153,6 +159,14 @@ def self.check_required_fields_present!(masks = MASKS)
       raise TransitionConfig::RequiredFieldsMissingException.new(missing) unless missing.empty?
     end
 
+    def self.check_homepage_protocol_present!(masks = MASKS)
+      missing = []
+      TransitionConfig::Site.all_with_basenames(masks).each do |site, _|
+        missing << site.abbr unless %w(http https).any? { |protocol| site.homepage.include?(protocol) }
```
Comment (issyl0) [inline lib/transition-config/site.rb]: I just tested this by adding "http" at the _end_ of a homepage URL and it didn't complain, so `include?` might be better as `start_with?`?

**Comment by brenetic** [comment]
Thanks @issyl0 have amended that

File: lib/transition-config/required_homepage_protocol_exception.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module TransitionConfig
+  class RequiredHomepageProtocolException < RuntimeError
+    attr_reader :missing
+    def initialize(missing)
+      @missing = missing
+    end
+
+    def to_s
```
Comment (jackscotti) [inline lib/transition-config/required_homepage_protocol_exception.rb]: Where are you using this method?

Comment (brenetic) [inline lib/transition-config/required_homepage_protocol_exception.rb]: Used in the site_test.rb. Seems ok . . . <_<  . . .  >_> . . .

Comment (jackscotti) [inline lib/transition-config/required_homepage_protocol_exception.rb]: I get an error when I run this in the irb console: 
```
"Sites missing homepage protocols: \n"\
"test string"
```

What is the `\` for?

Comment (brenetic) [inline lib/transition-config/required_homepage_protocol_exception.rb]: Multi line concatenation without using `+`. Only to stay consistent with the other Exception Handlers in this repo to be honest.

Comment (jackscotti) [inline lib/transition-config/required_homepage_protocol_exception.rb]: Found out: http://stackoverflow.com/questions/2337510/ruby-can-i-write-multi-line-string-with-no-concatenation
:+1:

Comment (issyl0) [inline lib/transition-config/required_homepage_protocol_exception.rb]: :+1:

**Pull request merged**